managers/backup-manager/BackupStatusPoller.js

```diff
@@ -0,0 +1,261 @@
+              instance_id: instance_guid,
+              repeatInterval: retryInterval,
+              type: CONST.BACKUP.TYPE.ONLINE
+            };
+            return retry(() => cf.serviceFabrikClient.scheduleBackup(options), {
+              maxAttempts: 3,
+              minDelay: 500
+            });
+          }
+        }) : operationStatusResponse
+      )
+      .catch(err => {
+        logger.error(`Caught error while checking for operation completion status:`, err);
+        throw err;
+      });
+  }
+  static doPostFinishOperation(operationStatusResponse, operationName, opts) {
+    return Promise
+      .try(() => eventmesh.apiServerClient.updateResource({
+        resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+        resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+        resourceId: opts.backup_guid,
+        status: {
+          'state': operationStatusResponse.state
+        }
+      }))
+      .then(() => this._logEvent(opts, operationName, operationStatusResponse))
+      .then(() => {
+        if (operationStatusResponse.operationTimedOut) {
+          const msg = `Deployment ${opts.instance_guid} ${operationName} with backup guid ${opts.backup_guid} exceeding timeout time
+        ${config.backup.backup_restore_status_poller_timeout / 1000 / 60} (mins). Stopping status check`;
+          logger.error(msg);
+        } else {
+          logger.info(`Instance ${opts.instance_guid} ${operationName} for backup guid ${opts.backup_guid} completed -`, operationStatusResponse);
+        }
+        return operationStatusResponse;
+      });
+  }
+
+  static _logEvent(opts, operation, operationStatusResponse) {
+    const eventLogger = EventLogInterceptor.getInstance(config.external.event_type, 'external');
+    const check_res_body = true;
+    const resp = {
+      statusCode: 200,
+      body: operationStatusResponse
+    };
+    if (CONST.URL[operation]) {
+      return eventLogger.publishAndAuditLogEvent(CONST.URL[operation], CONST.HTTP_METHOD.POST, opts, resp, check_res_body);
+    }
+  }
+
+  static start() {
+    function poller(object) {
+      const resourceDetails = eventmesh.apiServerClient.parseResourceDetailsFromSelfLink(object.metadata.selfLink);
+      // If no lockedByPoller annotation then set annotation  with time
+      // Else check timestamp if more than specific time than start polling and change lockedByPoller Ip
+      return eventmesh.apiServerClient.getResource({
+          resourceGroup: resourceDetails.resourceGroup,
+          resourceType: resourceDetails.resourceType,
+          resourceId: object.metadata.name,
+        })
+        .then(resourceBody => {
+          const options = resourceBody.spec.options;
+          const response = resourceBody.status.response;
+          const pollerAnnotation = resourceBody.metadata.annotations.lockedByTaskPoller;
+          logger.debug(`pollerAnnotation is ${pollerAnnotation} current time is: ${new Date()}`);
```

Comment:
can we update this to "Backup pollerAnnotation"

Reply:
Done